source/docs/essentials/extend-the-graphql-schema.md

```diff
@@ -0,0 +1,443 @@
+{
+  product(sku: "WH09") {
+    sku
+    clicksCounter
+  }
+}
+```
+
+Our GraphQL module does not yet have any code for sending content, so you must
+see the following response:
+
+```json
+{
+  "data": {
+    "product": {
+      "sku": "WH09",
+      "clicksCounter": null
+    }
+  }
+}
+```
+
+Let’s now add some executable logic to our module.
+
+### Implement resolvers
+
+The GraphQL middleware relies on resolver functions to determine the data to
+return for a given field. This is where most of the « real work » is done, for
+instance by fetching remote datasources or transforming data.
+
+Resolvers are exposed using the `resolvers` key of the module definition. It
+should be a **Resolver map**: an object where each key is a GraphQL type name,
+and values are mapping between field names and resolver function. Resolver
+functions may return a
+[Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises)
+for asynchronous operations.
+
+<blockquote class="more">
+  To know more about resolvers and their internals, we recommend the reading of
+  [GraphQL Tools resolvers
+  documentation](https://www.apollographql.com/docs/graphql-tools/resolvers).
+</blockquote>
+
+First, update the module definition as follow:
+
+```diff
+// my-module/server/modules/clicks-counters/index.js
+import typeDefs from "./schema.gql";
++import resolvers from "./resolvers";
+
+export default {
+  namespace: "ClicksCounters",
+-  typeDefs: typeDefs
++  typeDefs: typeDefs,
++  resolvers: resolvers
+};
+```
+
+And then create the `my-module/server/modules/clicks-counters/resolvers.js` file
+with a resolver map as below (we are going to analyze it in details in a few
+seconds):
+
+```js
+// my-module/server/modules/clicks-counters/resolvers.js
+const counters = new Map();
+
+const currentValueOf = sku => counters.get(sku) || 0;
+
+module.exports = {
+  Product: {
+    clicksCounter: ({ sku }) => currentValueOf(sku)
+  },
+
+  Mutation: {
+    incrementProductCounter(_, { sku, incrementValue = 1 }) {
+      counters.set(sku, currentValueOf(sku) + incrementValue);
+      return {
+        success: true
+      };
+    }
+  }
+};
+```
+
+Let’s analyze this code part by part to understand what is possible to achieve
+in a module.
+
+First of all, the exported resolver map defines a resolver for the
+`clicksCounter` field of any `Product` field.
+
+```js
+// …
+module.exports = {
+  Product: {
+    clicksCounter: ({ sku }) => currentValueOf(sku)
+  }
+  // …
+};
+```
+
+The resolver function will use the `sku` key from its parent data and will
+return the the current counter value from its local state (defaulting to `0` if
+no clicks occured).
+
+It is important to understand that the `sku` is not a parameter provided by
+frontend developers in the Query. It comes from data returned by earlier
+resolvers that fetched a `Product`, no matter where in the graph (category,
+upsells, cart…).
+
+<blockquote class="info">
```

Comment:
Sometimes we use note, sometimes we use info :man_shrugging:

Reply:
We’ll have to homogenize it when/if styling it. For now, I agree: :woman_shrugging: